Test failure ImportNativeAppEngineStandardProjectTest#importAppEngineStandardJava8_from1_3_1

Odd test failure seen on Travis on Oxygen build (for #501: no relevant code changes).

```
[Watchdog] > importAppEngineStandardJava8_from1_3_1(com.google.cloud.tools.eclipse.integration.appengine.ImportNativeAppEngineStandardProjectTest)
Window manager warning: Buggy client sent a _NET_ACTIVE_WINDOW message with a timestamp of 0 for 0x80317b (Import )
Window manager warning: meta_window_activate called by a pager with a 0 timestamp; the pager needs to be fixed.
16:41:11.295 [WorkbenchTestable] WARN  o.e.s.s.f.widgets.SWTBotTreeItem - Tree item TreeItem with text {General} is already expanded. Won't expand it again.
16:41:11.440 [main] WARN  o.e.s.s.finder.widgets.SWTBotButton - Widget is not enabled: (of type 'Button' and with mnemonic 'Next >' and with style 'SWT.PUSH')
16:41:11.445 [main] WARN  o.e.s.s.finder.widgets.SWTBotButton - Widget is not enabled: (of type 'Button' and with mnemonic 'Next >' and with style 'SWT.PUSH')
16:41:11.448 [main] WARN  o.e.s.s.finder.widgets.SWTBotButton - Widget is not enabled: (of type 'Button' and with mnemonic 'Next >' and with style 'SWT.PUSH')
16:41:11.452 [main] WARN  o.e.s.s.finder.widgets.SWTBotButton - Widget is not enabled: (of type 'Button' and with mnemonic 'Next >' and with style 'SWT.PUSH')
16:41:11.456 [main] WARN  o.e.s.s.finder.widgets.SWTBotButton - Widget is not enabled: (of type 'Button' and with mnemonic 'Next >' and with style 'SWT.PUSH')
16:41:12.055 [main] WARN  o.e.s.s.finder.widgets.SWTBotButton - Widget is not enabled: (of type 'Button' and with mnemonic 'Finish' and with style 'SWT.PUSH')
16:41:12.060 [main] WARN  o.e.s.s.finder.widgets.SWTBotButton - Widget is not enabled: (of type 'Button' and with mnemonic 'Finish' and with style 'SWT.PUSH')
16:41:12.067 [main] WARN  o.e.s.s.finder.widgets.SWTBotButton - Widget is not enabled: (of type 'Button' and with mnemonic 'Finish' and with style 'SWT.PUSH')
16:41:12.073 [main] WARN  o.e.s.s.finder.widgets.SWTBotButton - Widget is not enabled: (of type 'Button' and with mnemonic 'Finish' and with style 'SWT.PUSH')
16:41:12.078 [main] WARN  o.e.s.s.finder.widgets.SWTBotButton - Widget is not enabled: (of type 'Button' and with mnemonic 'Finish' and with style 'SWT.PUSH')
Window manager warning: Buggy client sent a _NET_ACTIVE_WINDOW message with a timestamp of 0 for 0x800021 (data - Ecl)
Window manager warning: meta_window_activate called by a pager with a 0 timestamp; the pager needs to be fixed.
ProjectUtils#waitForProjects[11.12 s]: waiting for 1 jobs: [Validation(1078)]
ProjectUtils#waitForProjects[11.18 s]: waiting for 1 jobs: [Building workspace(4)]
Window manager warning: Buggy client sent a _NET_ACTIVE_WINDOW message with a timestamp of 0 for 0x803635 (Delete Res)
Window manager warning: meta_window_activate called by a pager with a 0 timestamp; the pager needs to be fixed.
16:41:30.807 [main] WARN  o.e.s.s.finder.widgets.SWTBotButton - Widget is not enabled: (of type 'Button' and with mnemonic 'OK' and with style 'SWT.PUSH')
16:41:30.822 [main] WARN  o.e.s.s.finder.widgets.SWTBotButton - Widget is not enabled: (of type 'Button' and with mnemonic 'OK' and with style 'SWT.PUSH')
16:41:30.827 [main] WARN  o.e.s.s.finder.widgets.SWTBotButton - Widget is not enabled: (of type 'Button' and with mnemonic 'OK' and with style 'SWT.PUSH')
16:41:30.835 [main] WARN  o.e.s.s.finder.widgets.SWTBotButton - Widget is not enabled: (of type 'Button' and with mnemonic 'OK' and with style 'SWT.PUSH')
16:41:30.835 [main] WARN  o.e.s.s.finder.widgets.SWTBotButton - Widget is not enabled: (of type 'Button' and with mnemonic 'OK' and with style 'SWT.PUSH')
Window manager warning: Buggy client sent a _NET_ACTIVE_WINDOW message with a timestamp of 0 for 0x800021 (data - Ecl)
Window manager warning: meta_window_activate called by a pager with a 0 timestamp; the pager needs to be fixed.
Tests run: 2, Failures: 1, Errors: 0, Skipped: 0, Time elapsed: 60.493 sec <<< FAILURE! - in com.google.cloud.tools.eclipse.integration.appengine.ImportNativeAppEngineStandardProjectTest
importAppEngineStandardJava7_from1_3_1(com.google.cloud.tools.eclipse.integration.appengine.ImportNativeAppEngineStandardProjectTest)  Time elapsed: 38.532 sec
importAppEngineStandardJava8_from1_3_1(com.google.cloud.tools.eclipse.integration.appengine.ImportNativeAppEngineStandardProjectTest)  Time elapsed: 21.94 sec  <<< FAILURE!
java.lang.AssertionError: Imported App Engine standard Java 8 project has errors
/AESv8:-1: Faceted project metadata file "/AESv8/.settings/org.eclipse.wst.common.project.facet.core.xml" could not be read.
/AESv8/src/main/java/HelloAppEngine.java:8: WebServlet cannot be resolved to a type
/AESv8/src/main/java/HelloAppEngine.java:3: The import javax.servlet.annotation cannot be resolved
	at org.junit.Assert.fail(Assert.java:88)
	at org.junit.Assert.assertTrue(Assert.java:41)
	at com.google.cloud.tools.eclipse.test.util.project.ProjectUtils.failIfBuildErrors(ProjectUtils.java:185)
	at com.google.cloud.tools.eclipse.integration.appengine.ImportNativeAppEngineStandardProjectTest.importAppEngineStandardJava8_from1_3_1(ImportNativeAppEngineStandardProjectTest.java:101)
```

--- a/plugins/com.google.cloud.tools.eclipse.integration.appengine/src/com/google/cloud/tools/eclipse/integration/appengine/ImportNativeAppEngineStandardProjectTest.java
+++ b/plugins/com.google.cloud.tools.eclipse.integration.appengine/src/com/google/cloud/tools/eclipse/integration/appengine/ImportNativeAppEngineStandardProjectTest.java
@@ -66,7 +66,7 @@ public class ImportNativeAppEngineStandardProjectTest extends BaseProjectTest {
     updateOldContainers();
 
     ProjectUtils.waitForProjects(project);
-    ProjectUtils.waitUntilNoBuildError(project);
+    ProjectUtils.waitUntilNoBuildErrors(project);
 
     IFacetedProject facetedProject = ProjectFacetsManager.create(project);
     assertNotNull("should be a faceted project", facetedProject);
@@ -94,7 +94,7 @@ public class ImportNativeAppEngineStandardProjectTest extends BaseProjectTest {
     updateOldContainers();
 
     ProjectUtils.waitForProjects(project);
-    ProjectUtils.waitUntilNoBuildError(project);
+    ProjectUtils.waitUntilNoBuildErrors(project);
 
     IFacetedProject facetedProject = ProjectFacetsManager.create(project);
     assertNotNull("should be a faceted project", facetedProject);

--- a/plugins/com.google.cloud.tools.eclipse.integration.appengine/src/com/google/cloud/tools/eclipse/integration/appengine/ImportNativeAppEngineStandardProjectTest.java
+++ b/plugins/com.google.cloud.tools.eclipse.integration.appengine/src/com/google/cloud/tools/eclipse/integration/appengine/ImportNativeAppEngineStandardProjectTest.java
@@ -66,8 +66,7 @@ public class ImportNativeAppEngineStandardProjectTest extends BaseProjectTest {
     updateOldContainers();
 
     ProjectUtils.waitForProjects(project);
-    ProjectUtils.failIfBuildErrors("Imported App Engine standard Java 7 project has errors",
-        project);
+    ProjectUtils.waitUntilNoBuildError(project);
 
     IFacetedProject facetedProject = ProjectFacetsManager.create(project);
     assertNotNull("should be a faceted project", facetedProject);
@@ -95,8 +94,7 @@ public class ImportNativeAppEngineStandardProjectTest extends BaseProjectTest {
     updateOldContainers();
 
     ProjectUtils.waitForProjects(project);
-    ProjectUtils.failIfBuildErrors("Imported App Engine standard Java 8 project has errors",
-        project);
+    ProjectUtils.waitUntilNoBuildError(project);
 
     IFacetedProject facetedProject = ProjectFacetsManager.create(project);
     assertNotNull("should be a faceted project", facetedProject);

--- a/plugins/com.google.cloud.tools.eclipse.test.util/src/com/google/cloud/tools/eclipse/test/util/project/ProjectUtils.java
+++ b/plugins/com.google.cloud.tools.eclipse.test.util/src/com/google/cloud/tools/eclipse/test/util/project/ProjectUtils.java
@@ -142,7 +142,7 @@ public class ProjectUtils {
     // wait for any post-import operations too
     waitForProjects(projects);
     if (checkBuildErrors) {
-      waitUntilNoBuildError();
+      waitUntilNoBuildErrors();
       // changed from specific projects to see all possible errors
       failIfBuildErrors("Imported projects have errors");
     }
@@ -150,12 +150,12 @@ public class ProjectUtils {
     return projects;
   }
 
-  public static void waitUntilNoBuildError() throws CoreException {
+  public static void waitUntilNoBuildErrors() throws CoreException {
     IProject[] projects = getWorkspace().getRoot().getProjects();
-    waitUntilNoBuildError(projects);
+    waitUntilNoBuildErrors(projects);
   }
 
-  public static void waitUntilNoBuildError(IProject... projects) throws CoreException {
+  public static void waitUntilNoBuildErrors(IProject... projects) throws CoreException {
     try {
       Stopwatch elapsed = Stopwatch.createStarted();
       while (true) {

--- a/plugins/com.google.cloud.tools.eclipse.test.util/src/com/google/cloud/tools/eclipse/test/util/project/ProjectUtils.java
+++ b/plugins/com.google.cloud.tools.eclipse.test.util/src/com/google/cloud/tools/eclipse/test/util/project/ProjectUtils.java
@@ -150,11 +150,16 @@ public class ProjectUtils {
     return projects;
   }
 
-  private static void waitUntilNoBuildError() throws CoreException {
+  public static void waitUntilNoBuildError() throws CoreException {
+    IProject[] projects = getWorkspace().getRoot().getProjects();
+    waitUntilNoBuildError(projects);
+  }
+
+  public static void waitUntilNoBuildError(IProject... projects) throws CoreException {
     try {
       Stopwatch elapsed = Stopwatch.createStarted();
       while (true) {
-        Set<String> errors = getAllBuildErrors();
+        Set<String> errors = getAllBuildErrors(projects);
         if (errors.isEmpty() || elapsed.elapsed(TimeUnit.SECONDS) > 300) {
           return;
         }

